src/sentry/runner/commands/createuser.py

```diff
@@ -76,8 +76,14 @@ def createuser(email, password, superuser, no_password, no_input, force_update):
     if password:
         user.set_password(password)
 
-    if User.objects.filter(username=email).exists():
+    try:
+        existing_user_id = User.objects.get(username=email).id
+    except User.DoesNotExist:
+        existing_user_id = None
+
+    if existing_user_id is not None:
         if force_update:
+            user.id = existing_user_id
             user.save(force_update=force_update)
```

Comment:
I don't think this is what you want to do:

```
In [25]: user_3 = User.objects.create(username='test_tesjtst', email='test@test.com')
In [28]: user_3.set_password('hello')
In [30]: (user_3.username, user_3.email, user_3.password)
Out[30]:
('test_tesjtst',
 'test@test.com',
 'pbkdf2_sha256$36000$pK6zZ1CGK7Qu$Pcm9GpEINVg5lJGL/SDh0k9qR9qnpqcIQdTXMivad6U=')

In [31]: some_user = User(email='hi@bye.com', username='hi@by.com')
In [32]: some_user.id = user_3.id
In [33]: some_user.save()

In [34]: user_3.refresh_from_db()
In [35]: (user_3.username, user_3.email, user_3.password)
Out[35]: ('hi@by.com', 'hi@bye.com', '')
```

This overwrites all parameters on the existing row with what's in the new `User`, which means nulling out anything that you didn't specify. You should probably instead user `existing_user.update(<fields you want to update>)`

Reply:
Okay, refactored around this.